src/jit/morph.cpp

```diff
@@ -7411,6 +7411,41 @@ void Compiler::fgMorphTailCallViaHelper(GenTreeCall* call, void* pfnCopyArgs)
     if (call->IsVirtualStub())
     {
         call->gtFlags |= GTF_CALL_NULLCHECK;
+
+#if defined(_TARGET_AMD64_)
+        // If we already inited arg info here then we will have added the VSD
+        // arg on AMD64. So we remove it here as we will handle this case
+        // specially below.
+        fgArgInfo* argInfo = call->fgArgInfo;
+        if (argInfo != nullptr)
```

Comment:
It should be safe to assert here that argInfo != nullptr. If we go down the path of morphing the tailcall via helper we have to have checked if it was a fast tail call and setup argInfo there.